src/main/java/ch/uzh/ifi/hase/soprafs24/service/AuthService.java

```diff
@@ -64,15 +64,17 @@ private HashMap<String,String> getSpotifyUserData(String accessToken){
         return SpotifyService.getUserData(accessToken);
     }
 
-    private User createOrLogin(String spotifyUserId, String username, SpotifyJWT spotifyJWT) {
+    private User createOrLogin(String spotifyUserId, String username, String image_url, SpotifyJWT spotifyJWT) {
         User userToCreateOrLogin = new User();
         userToCreateOrLogin.setSpotifyUserId(spotifyUserId);
         userToCreateOrLogin.setUsername(username);
+        userToCreateOrLogin.setImageUrl(image_url);
```

Comment:
We get the image every time user login? is the issue with image url expiration solved by this?

Reply:
Yes I made it in a way that we get the image (and also the username) at each login. With that we should be safe regarding changing urls and can even take care of changing pictures/names.